src/ImageSharp/Formats/Tiff/Compression/Decompressors/T4BitReader.cs

```diff
@@ -396,55 +379,106 @@ public void ReadNextRun()
             this.isFirstScanLine = false;
         }
 
-        public void StartNewRow()
+        /// <summary>
+        /// Initialization for a new row.
+        /// </summary>
+        public virtual void StartNewRow()
         {
             // Each new row starts with a white run.
             this.IsWhiteRun = true;
             this.isStartOfRow = true;
             this.terminationCodeFound = false;
+        }
 
-            if (this.isModifiedHuffmanRle)
+        /// <inheritdoc/>
+        public void Dispose() => this.Data.Dispose();
+
+        /// <summary>
+        /// An EOL is expected before the first data.
+        /// </summary>
+        protected virtual void ReadEolBeforeFirstData()
+        {
+            if (this.isFirstScanLine)
             {
-                int pad = 8 - (this.bitsRead % 8);
-                if (pad != 8)
+                this.Value = this.ReadValue(this.eolPadding ? 16 : 12);
+
+                if (!this.IsEndOfScanLine)
                 {
-                    // Skip padding bits, move to next byte.
-                    this.position++;
-                    this.bitsRead = 0;
+                    TiffThrowHelper.ThrowImageFormatException("ccitt compression parsing error: expected start of data marker not found");
                 }
+
+                this.Reset();
             }
         }
 
-        /// <inheritdoc/>
-        public void Dispose() => this.Data.Dispose();
+        /// <summary>
+        /// Resets the current value read and the number of bits read.
+        /// </summary>
+        /// <param name="resetRunLength">if set to true resets also the run length.</param>
+        protected void Reset(bool resetRunLength = true)
+        {
+            this.Value = 0;
+            this.CurValueBitsRead = 0;
+
+            if (resetRunLength)
+            {
+                this.RunLength = 0;
+            }
+        }
+
+        /// <summary>
+        /// Resets the bits read to 0.
+        /// </summary>
+        protected void ResetBitsRead() => this.BitsRead = 0;
+
+        /// <summary>
+        /// Reads the next value.
+        /// </summary>
+        /// <param name="nBits">The number of bits to read.</param>
+        /// <returns>The value read.</returns>
+        protected uint ReadValue(int nBits)
+        {
+            Guard.MustBeGreaterThan(nBits, 0, nameof(nBits));
```

Comment:
How often is this called. The Guard might be a performance killer

Reply:
its called very often, its `DebugGuard` now